### PR TITLE
Core Platform and persistence ignorance

### PR DESCRIPTION
**Pull request by TeganAli**
# About

This is a post about...

# Release

Is this post for a specific release? If so which one?

# Blog image idea

The design team will work on a design for the blog image. Add any ideas you have about the graphic image here.

# Pre-requisites

- [ ] The draft is complete and this post is ready to be reviewed.
- [ ] I'm familiar with the [writing for Octopus TL;DR](https://www.octopus.design/writing/writing-tldr/).
- [ ] The PR build passes validation, and if it doesn't, I've checked the [common validations errors](https://style.octopus.com/writing-at-octopus-tldr#common-validation-errors) and none of those apply.
- [ ] The images I've included follow the [rules for working with images](https://www.octopus.design/writing/working-with-images/).
- [ ] I've added alt text for all the images I've included (alt text describes image to people unable to see it, 125 characters max)
- [ ] One screenshot (if any are included) is suitable for Twitter. This screenshot should be 16:9 ratio and at least 1200 x 675 pixels. For screenshots from the Octopus Web Portal, please use Chrome and the browser’s zoom function to remove as much white space as possible. If screenshots are annotated (e.g., with boxes or arrows), please also provide a version that is unannotated for use on Twitter. Do not use dark mode for the Octopus Web Portal.

### CODE REVIEW

File: blog/2022-03/config-as-code-persistence-ignorance/index.md [hunk 1]
```diff
@@ -0,0 +1,340 @@
...
+
+Also, transactions are an advanced topic. Developers in our growing team could easily hold transactions in questionable ways or forget to commit entirely. Things like this would likely be caught during testing, but this illustrates the *unnecessary churn* that starts to creep in.
+
+Our job architecturally is to help our developers fall into the pit of success. When we observe complexities like this creeping in (especially as we start to scale), it's a great opportunity to consider a new abstraction and try to improve on the amazing foundations that got us here.
+
+## Introducing Unit of Work
+
+To help us manage transactions across requests/commands, we introduced a [Unit of Work](https://martinfowler.com/eaaCatalog/unitOfWork.html) so there'd be one less thing for developers to consider. Consumers just ask for a document store of a given type. The Unit of Work *surrounding* their operation would be managed for them and not their concern.
+
+We'll cover this topic in more detail in a follow-up post, but it's safe to say, consumers just want an `X`, so let them ask for *just the thing they need*.
```
Comment (TeganAli) [inline blog/2022-03/config-as-code-persistence-ignorance/index.md:210]: @robpearson should I delete mention of "We'll cover this topic in more detail in a follow-up post" ? I don't think we're publishing anything else related to this anymore.

File: blog/2022-03/config-as-code-persistence-ignorance/index.md [hunk 2]
```diff
@@ -0,0 +1,340 @@
...
+    ...
+```
+
+When you ask for a document with a given `id`, the `TryGetFromCache` method figures out whether the type supports caching and returns the result from the cache if it's available.
+
+The consumer knows nothing about this layer. We decorated our abstraction and put strong test coverage around *just this layer*, giving us confidence it does what we expect. Any consumers of `IDocumentStore<T>` will get these decorations out-of-the-box.
+
+## Encouraged by convention
+
+Our developers are now encouraged to take dependencies on `IDocumentStore<T>` and we have [convention tests](https://github.com/andrewabest/Conventional) to help them stop taking dependencies on stores/transactions directly (we'll have more on convention testing in a follow-up post).
```
Comment (TeganAli) [inline blog/2022-03/config-as-code-persistence-ignorance/index.md:298]: @robpearson should I delete mention of "we'll have more on convention testing in a follow-up post" ? I don't think we're publishing anything else related to this anymore.

Comment (robpearson) [reply blog/2022-03/config-as-code-persistence-ignorance/index.md:298]: Yes, we'd want to delete that. 👌

File: blog/2022-03/config-as-code-persistence-ignorance/index.md (hunk 1 above)
Comment (robpearson) [reply blog/2022-03/config-as-code-persistence-ignorance/index.md]: Eek. Apologise. I missed this one. Yes, we should delete this reference. 👍

Comment (TeganAli) [reply blog/2022-03/config-as-code-persistence-ignorance/index.md]: All good, thanks, @robpearson. I deleted it when you commented on the other one 😁

**Review by MarkSiedle — APPROVED** [review]
Not sure if this is waiting on me, but LGTM :)

**Pull request merged**